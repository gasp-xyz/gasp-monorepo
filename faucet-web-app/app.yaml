runtime: nodejs18
service: react-faucet

handlers:
  - url: /static/js/(.*)
    static_files: build/static/js/\1
    upload: build/static/js/(.*)
    secure: always
  - url: /static/css/(.*)
    static_files: build/static/css/\1
    upload: build/static/css/(.*)
    secure: always
  - url: /static/media/(.*)
    static_files: build/static/media/\1
    upload: build/static/media/(.*)
    secure: always
  - url: /(.*\.(json|ico|png|jpg|svg))$
    static_files: build/\1
    upload: build/.*\.(json|ico|png|jpg|svg)$
    secure: always
  - url: /
    static_files: build/index.html
    upload: build/index.html
    secure: always
  - url: /.*
    static_files: build/index.html
    upload: build/index.html
    secure: always