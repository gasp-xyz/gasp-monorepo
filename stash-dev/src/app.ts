import '@mangata-finance/types'
import express, { Request, Response, json, urlencoded } from 'express'
import swaggerUI from 'swagger-ui-express'
import cors from 'cors'

import * as priceDiscoveryController from './controller/PriceDiscoveryController.js'
import * as tokenController from './controller/TokenController.js'
import * as xcmController from './controller/XcmController.js'
import * as coingeckoController from './controller/coingeckoListingController.js'
import * as coinmarketcapController from './controller/coinmarketcapListingController.js'
import * as tokenListController from './controller/tokenListController.js'
import * as stakingAprController from './controller/stakingAprController.js'
import * as liquidStakingController from './controller/liquidStakingController.js'
import * as networkController from './controller/networkController.js'
import * as tokenNetworkPortfolioController from './controller/tokenNetworkPortfolioController.js'

import { createRequire } from 'module'
const require = createRequire(import.meta.url)
const swaggerFile = require('../swagger-output.json')

const app = express()

// App configuration
app.set('port', process.env.PORT || 8080)
app.use(json())
app.use(urlencoded({ extended: true }))
app.use(cors())

//Root (Health Check)
app.get('/', async (req: Request, res: Response): Promise<void> => {
  res.json({})
})

// Tokens
app.get('/token/order-buckets', tokenController.listTokenOrderBuckets)
// Token list
app.get('/token/list/stats', tokenListController.tokenList)
app.get('/token/:symbol/stats', tokenListController.tokenDetails)

// XCM
app.get('/xcm/channels', xcmController.channels)
app.get('/xcm/network', xcmController.network)
app.get('/xcm/tokens', xcmController.tokens)

app.get(
  '/collator/:collatorAddress/staking/dailyReward',
  stakingAprController.dailyReward
)
app.get('/collator/:collatorAddress/staking/apy', stakingAprController.apy)
app.get('/collators/staking/apy', stakingAprController.collatorsApy)

app.get(
  '/account/:address/liquid-staking/rewards-history/24h/sum',
  liquidStakingController.rewards24hours
)
app.get(
  '/account/:address/liquid-staking/rewards-history/month/sum',
  liquidStakingController.rewardsMonth
)

// Price Discovery API
app.get('/price-discovery/:currencySymbol/', priceDiscoveryController.getPrice)
app.get(
  '/price-history/pair/:baseCurrencySymbol/:targetCurrencySymbol',
  priceDiscoveryController.getHistoryPair
)
app.get('/price-history/:currencySymbol/', priceDiscoveryController.getHistory)
app.get(
  '/volume-history/pools/:currencySymbol/',
  priceDiscoveryController.getTradesPool
)
app.get(
  '/volume-history/:currencySymbol/',
  priceDiscoveryController.getTradesAsset
)
app.get(
  '/tvl-history/pools/:currencySymbol/',
  priceDiscoveryController.getVolumePool
)
app.get(
  '/tvl-history/:currencySymbol/',
  priceDiscoveryController.getVolumeAsset
)

// CoinGecko listing endpoints
app.get('/coingecko/pairs', coingeckoController.pairs)
app.get('/coingecko/tickers', coingeckoController.tickers)

if (process.env.APP_ENV === 'rollup-dev') {
  app.get('/affirmed-network/list', networkController.networkList)
  app.get('/affirmed-token/list', networkController.tokenList)
  app.get(
    '/account/:address/token-portfolio',
    tokenNetworkPortfolioController.tokenNetworkPortfolio
  )
}

// Coinmarketcap listing endpoints
app.get('/coinmarketcap/v1/summary', coinmarketcapController.summary)

app.use('/doc', swaggerUI.serve, swaggerUI.setup(swaggerFile))

export default app
