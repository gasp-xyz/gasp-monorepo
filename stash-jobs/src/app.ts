import 'gasp-types'
import express, { Request, Response, json, urlencoded } from 'express'
import cors from 'cors'

import { createRequire } from 'module'
const require = createRequire(import.meta.url)
const swaggerFile = require('../swagger-output.json')

const app = express()

// App configuration
app.set('port', process.env.PORT || 8080)
app.use(json())
app.use(urlencoded({ extended: true }))
app.use(cors())

//Root (Health Check)
app.get('/', async (req: Request, res: Response): Promise<void> => {
  res.json({})
})
export default app
