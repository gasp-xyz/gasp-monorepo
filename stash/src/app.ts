import 'gasp-types'
import express, { Request, Response, json, urlencoded } from 'express'
import swaggerUI from 'swagger-ui-express'
import cors from 'cors'

import * as priceDiscoveryController from './controller/PriceDiscoveryController.js'
import * as tokenController from './controller/TokenController.js'
import * as xcmController from './controller/XcmController.js'
import * as coingeckoController from './controller/coingeckoListingController.js'
import * as coinmarketcapController from './controller/coinmarketcapListingController.js'
import * as tokenListController from './controller/tokenListController.js'
import * as stakingAprController from './controller/stakingAprController.js'
import * as liquidStakingController from './controller/liquidStakingController.js'
import * as networkController from './controller/networkController.js'
import * as tokenNetworkPortfolioController from './controller/tokenNetworkPortfolioController.js'
import * as faucetController from './controller/FaucetController.js'
import * as tracingController from './controller/TracingController.js'

import { createRequire } from 'module'
const require = createRequire(import.meta.url)
const swaggerFile = require('../swagger-output.json')

const app = express()

// App configuration
app.set('port', process.env.PORT || 8080)
app.use(json())
app.use(urlencoded({ extended: true }))
app.use(cors())

//Root (Health Check)
app.get('/', async (req: Request, res: Response): Promise<void> => {
  res.json({})
})

// Tokens
app.get('/token/order-buckets', tokenController.listTokenOrderBuckets)
// Token list
app.get('/token/list/stats', tokenListController.tokenList)
app.get('/token/:id/stats', tokenListController.tokenDetails)

// XCM
app.get('/xcm/channels', xcmController.channels)
app.get('/xcm/network', xcmController.network)
app.get('/xcm/tokens', xcmController.tokens)

app.get(
  '/collator/:collatorAddress/staking/dailyReward',
  stakingAprController.dailyReward
)
app.get('/collator/:collatorAddress/staking/apy', stakingAprController.apy)
app.get('/collators/staking/apy', stakingAprController.collatorsApy)

app.get(
  '/account/:address/liquid-staking/rewards-history/24h/sum',
  liquidStakingController.rewards24hours
)
app.get(
  '/account/:address/liquid-staking/rewards-history/month/sum',
  liquidStakingController.rewardsMonth
)

// Price Discovery API
app.get('/price-discovery/:currencyId/', priceDiscoveryController.getPrice)
app.get(
  '/price-history/pair/:baseCurrencyId/:targetCurrencyId',
  priceDiscoveryController.getHistoryPair
)
app.get('/price-history/:currencyId/', priceDiscoveryController.getHistory)
app.get(
  '/volume-history/pools/:currencyId/',
  priceDiscoveryController.getTradesPool
)
app.get('/volume-history/:currencyId/', priceDiscoveryController.getTradesAsset)
app.get(
  '/tvl-history/pools/:currencyId/',
  priceDiscoveryController.getVolumePool
)
app.get('/tvl-history/:currencyId/', priceDiscoveryController.getVolumeAsset)

// CoinGecko listing endpoints
app.get('/coingecko/pairs', coingeckoController.pairs)
app.get('/coingecko/tickers', coingeckoController.tickers)

app.get('/affirmed-network/list', networkController.networkList)
app.get('/affirmed-token/list', networkController.tokenList)
app.get(
  '/account/:address/token-portfolio',
  tokenNetworkPortfolioController.tokenNetworkPortfolio
)

//Faucet endpoint
app.get(
  '/faucet/requestTokens/:toAddress/captcha/:captcha/',
  faucetController.captcha
)

// Tracing endpoints
app.post('/tracing/tx/start', tracingController.startTracing)

app.get(
  '/tracing/tx/:txHashOrEntityId',
  tracingController.getTransactionStatusByTxHashOrEntityId
)

app.get(
  '/tracing/tx/listByAddress/:address',
  tracingController.getAllTransactionsByAddress
)

app.get(
  '/tracing/tx/listByAddress/:address/:status',
  tracingController.getAllTransactionsByAddressAndStatus
)

app.get(
  '/tracing/tx/findByEntityId/:entityId',

  tracingController.getATransactionByEntityId
)

// Coinmarketcap listing endpoints
app.get('/coinmarketcap/v1/summary', coinmarketcapController.summary)

app.use('/doc', swaggerUI.serve, swaggerUI.setup(swaggerFile))

export default app
