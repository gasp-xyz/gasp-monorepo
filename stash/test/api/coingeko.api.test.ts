/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { describe, expect, it } from 'vitest'
import supertest from 'supertest'
import app from '../../src/app'
import Joi from 'joi'

const pairsSchema = Joi.array().items(
  Joi.object({
    ticker_id: Joi.string().required(),
    base: Joi.string().required(),
    target: Joi.string().required(),
    pool_id: Joi.number().required(),
  })
)
const tickersSchema = Joi.array().items(
  Joi.object({
    ticker_id: Joi.string().required(),
    base_currency: Joi.string().required(),
    target_currency: Joi.string().required(),
    last_price: Joi.number().precision(18).unsafe().required(),
    base_volume: Joi.number().precision(18).unsafe().required(),
    target_volume: Joi.number().precision(18).unsafe().required(),
    pool_id: Joi.number().precision(0).required(),
    liquidity_in_usd: Joi.number().precision(18).unsafe().required(),
  })
)

describe('APi tests: Coingeko', () => {
  it('GET /pairs - Schema validation', async () => {
    await supertest(app)
      .get('/coingecko/pairs')
      .expect(200)
      .then((response) => {
        const validationResult = pairsSchema.validate(response.body)
        expect(validationResult.error).toBeUndefined()
      })
  })
  it('GET /tickers - Schema validation', async () => {
    await supertest(app)
      .get('/coingecko/tickers')
      .expect(200)
      .then((response) => {
        const validationResult = tickersSchema.validate(response.body)
        expect(validationResult.error).toBeUndefined()
      })
  })
  it('GET tickers & GET pairs pool info matches', async () => {
    await supertest(app)
      .get('/coingecko/tickers')
      .expect(200)
      .then(async (response) => {
        const pairs = (await supertest(app).get('/coingecko/pairs').expect(200))
          .body
        const tickers = response.body
        expect(tickers.length).toEqual(pairs.length)
        tickers.forEach((ticker) => {
          expect(
            pairs.filter(
              (pair) =>
                pair.pool_id === ticker.pool_id &&
                ticker.ticker_id === pair.ticker_id &&
                ticker.base_currency === pair.base &&
                ticker.target_currency === pair.target
            )
          ).toHaveLength(1)
        })
      })
  })
})

// These tests will fail if images changes And/Or if bugfixes. Careful when updating!
// UPDATE:
//    /pairs connects to Kusama network, so toMatchSnapshot may fail from
//    time to time until we do MGX-792
describe('Snapshots tests: Coingeko', () => {
  it('GET /pairs - Schema validation', async () => {
    await supertest(app)
      .get('/coingecko/pairs')
      .expect(200)
      .then((response) => {
        expect(response.body).toMatchSnapshot('Pairs')
      })
  })
  it('GET /tickers - Schema validation', async () => {
    await supertest(app)
      .get('/coingecko/tickers')
      .expect(200)
      .then((response) => {
        response.body
          .sort((a, b) => a.pool_id - b.pool_id)
          .forEach((ticker) => {
            expect(ticker).toMatchSnapshot({
              liquidity_in_usd: expect.any(String),
              target_volume: expect.any(String),
              base_volume: expect.any(String),
              last_price: expect.any(String),
            })
          })
      })
  })
})
describe('API Errors: Coingeko', () => {
  it('GET non existing path', async () => {
    await supertest(app).get('/coingecko/foo').expect(404)
  })
  it('POST /tickers ', async () => {
    await supertest(app).post('/coingecko/tickers').expect(404)
  })
})

//Need to skip this, since we re not able to stop the container.
// Unless we use a different container for each test suite we need to skip this.
// Using a different container for each test suite is not possible,
// since resolution of redis-host is done in the import.
describe.skip('System Errors: Coingeko', () => {
  it('GET /tickers  with redis down', async () => {
    // await container.stop();
    await supertest(app).get('/coingecko/tickers').expect(500)
  })
})
