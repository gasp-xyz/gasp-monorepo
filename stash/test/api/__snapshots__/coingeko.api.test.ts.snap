// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots tests: Coingeko > GET /pairs - Schema validation > Pairs 1`] = `
[
  {
    "base": "GOLD",
    "pool_id": "5",
    "target": "GETH",
    "ticker_id": "GOLD_GETH",
  },
  {
    "base": "SLVR",
    "pool_id": "7",
    "target": "COPP",
    "ticker_id": "SLVR_COPP",
  },
  {
    "base": "GASP",
    "pool_id": "10",
    "target": "GETH",
    "ticker_id": "GASP_GETH",
  },
  {
    "base": "USDT",
    "pool_id": "12",
    "target": "GETH",
    "ticker_id": "USDT_GETH",
  },
  {
    "base": "SLVR",
    "pool_id": "15",
    "target": "PLAT",
    "ticker_id": "SLVR_PLAT",
  },
  {
    "base": "DEEPSEEK AI",
    "pool_id": "20",
    "target": "PEPE",
    "ticker_id": "DEEPSEEK AI_PEPE",
  },
  {
    "base": "PEPE",
    "pool_id": "21",
    "target": "LINK",
    "ticker_id": "PEPE_LINK",
  },
  {
    "base": "GETH",
    "pool_id": "22",
    "target": "SLVR",
    "ticker_id": "GETH_SLVR",
  },
  {
    "base": "PLAT",
    "pool_id": "23",
    "target": "GOLD",
    "ticker_id": "PLAT_GOLD",
  },
  {
    "base": "LINK",
    "pool_id": "24",
    "target": "GOLD",
    "ticker_id": "LINK_GOLD",
  },
]
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 1`] = `
{
  "base_currency": "GOLD",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "5",
  "target_currency": "GETH",
  "target_volume": Any<String>,
  "ticker_id": "GOLD_GETH",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 2`] = `
{
  "base_currency": "SLVR",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "7",
  "target_currency": "COPP",
  "target_volume": Any<String>,
  "ticker_id": "SLVR_COPP",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 3`] = `
{
  "base_currency": "GASP",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "10",
  "target_currency": "GETH",
  "target_volume": Any<String>,
  "ticker_id": "GASP_GETH",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 4`] = `
{
  "base_currency": "USDT",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "12",
  "target_currency": "GETH",
  "target_volume": Any<String>,
  "ticker_id": "USDT_GETH",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 5`] = `
{
  "base_currency": "SLVR",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "15",
  "target_currency": "PLAT",
  "target_volume": Any<String>,
  "ticker_id": "SLVR_PLAT",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 6`] = `
{
  "base_currency": "DEEPSEEK AI",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "20",
  "target_currency": "PEPE",
  "target_volume": Any<String>,
  "ticker_id": "DEEPSEEK AI_PEPE",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 7`] = `
{
  "base_currency": "PEPE",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "21",
  "target_currency": "LINK",
  "target_volume": Any<String>,
  "ticker_id": "PEPE_LINK",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 8`] = `
{
  "base_currency": "GETH",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "22",
  "target_currency": "SLVR",
  "target_volume": Any<String>,
  "ticker_id": "GETH_SLVR",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 9`] = `
{
  "base_currency": "PLAT",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "23",
  "target_currency": "GOLD",
  "target_volume": Any<String>,
  "ticker_id": "PLAT_GOLD",
}
`;

exports[`Snapshots tests: Coingeko > GET /tickers - Schema validation 10`] = `
{
  "base_currency": "LINK",
  "base_volume": Any<String>,
  "last_price": Any<String>,
  "liquidity_in_usd": Any<String>,
  "pool_id": "24",
  "target_currency": "GOLD",
  "target_volume": Any<String>,
  "ticker_id": "LINK_GOLD",
}
`;
